src/kiota/Extension/KiotaHostExtensions.cs

```diff
@@ -93,5 +96,33 @@ static void ConfigureServiceContainer(HostBuilderContext context, IServiceCollec
 
             return OperatingSystem.IsFreeBSD() ? "freebsd" : null;
         }
+
+        static string AcquisitionChannel(string? path)
+        {
+            if (path != null && !string.IsNullOrWhiteSpace(path))
+            {
+                var absolutePath = Path.GetFullPath(path);
+                var homeDir = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+                // Dotnet tool
+                if (absolutePath.StartsWith(Path.Join(homeDir, ".dotnet", "tools")))
```

Comment:
docker should be covered with environment variables.
vscode extension should be known from the .vscode in the path